google-cloud-firestore/src/main/java/com/google/cloud/firestore/QuerySnapshot.java

```diff
@@ -104,134 +49,16 @@ public static QuerySnapshot withChanges(
-        return Collections.unmodifiableList(documentChanges);
-      }
-
-      @Override
-      public int size() {
-        return documentSet.size();
-      }
-
-      @Override
-      public boolean equals(Object o) {
-        if (this == o) {
-          return true;
-        }
-        if (o == null || getClass() != o.getClass()) {
-          return false;
-        }
-        QuerySnapshot that = (QuerySnapshot) o;
-        return Objects.equals(query, that.query)
-            && Objects.equals(this.size(), that.size())
-            && Objects.equals(this.getDocumentChanges(), that.getDocumentChanges())
-            && Objects.equals(this.getDocuments(), that.getDocuments());
-      }
-
-      @Override
-      public int hashCode() {
-        return Objects.hash(query, this.getDocumentChanges(), this.getDocuments());
-      }
-    };
+    return new QuerySnapshot(query, readTime, null, documentSet, documentChanges);
   }
 
   /**
    * Returns the query for the snapshot.
    *
    * @return The backing query that produced this snapshot.
    */
-  @Nonnull
-  public Query getQuery() {
-    return query;
-  }
-
-  /**
-   * Returns the time at which this snapshot was read.
-   *
-   * @return The read time of this snapshot.
-   */
-  @Nonnull
-  public Timestamp getReadTime() {
-    return readTime;
-  }
-
-  /**
-   * Returns the documents in this QuerySnapshot as a List in order of the query.
-   *
-   * @return The list of documents.
-   */
-  @Nonnull
-  public abstract List<QueryDocumentSnapshot> getDocuments();
-
-  /**
-   * Returns the list of documents that changed since the last snapshot. If it's the first snapshot
-   * all documents will be in the list as added changes.
-   *
-   * @return The list of documents that changed since the last snapshot.
-   */
-  @Nonnull
-  public abstract List<DocumentChange> getDocumentChanges();
-
-  /** Returns true if there are no documents in the QuerySnapshot. */
-  public boolean isEmpty() {
-    return this.size() == 0;
-  }
-
-  /** Returns the number of documents in the QuerySnapshot. */
-  public abstract int size();
-
   @Override
-  @Nonnull
-  public Iterator<QueryDocumentSnapshot> iterator() {
-    return getDocuments().iterator();
-  }
-
-  /**
-   * Returns the contents of the documents in the QuerySnapshot, converted to the provided class, as
-   * a list.
-   *
-   * @param clazz The POJO type used to convert the documents in the list.
-   */
-  @Nonnull
-  public <T> List<T> toObjects(@Nonnull Class<T> clazz) {
-    List<QueryDocumentSnapshot> documents = getDocuments();
-    List<T> results = new ArrayList<>(documents.size());
-    for (DocumentSnapshot documentSnapshot : documents) {
-      results.add(
-          CustomClassMapper.convertToCustomClass(
-              documentSnapshot.getData(), clazz, documentSnapshot.getReference()));
-    }
-
-    return results;
+  public Query getQuery() {
+    return super.getQuery();
   }
```

Comment:
This `Override` doesn't seem to do anything and could just be removed. It only serves to remove `@Nonnull` annotation from super. Is that intentional?

Reply:
Without this, the clirr style check fails reporting ` Method 'public com.google.cloud.firestore.Query getQuery()' has been removed`. It's effectively the same, but `getQuery()` on the base is generic.

Reply:
Following up on our discussion offline, I removed both of these overrides.